Sources/Fluent/Database/Transactable.swift

```diff
@@ -11,12 +11,12 @@ extension Database: Transactable {
     /// be rolled back.
     ///
     /// note: the underlying driver must support transactions
-    public func transaction(_ closure: (Connection) throws -> ()) throws {
+    public func transaction<R>(_ closure: (Connection) throws -> R) throws -> R {
```

Comment:
As far as I know it won't be necessary when `R` is inferred to `Void` - and this way the compiler will emit the warning is unused and `R` isn't `Void` - which I would expect it to be the desired behavior?

This has been copied off `DispatchQueue.sync<T>(execute work: () throws -> T) rethrows -> T`.